examples/discord_bot.py

```diff
@@ -204,34 +243,43 @@ async def current_war_status(ctx, clan_tag):
 
         e.add_field(name=war.clan.name, value=war.clan.tag)
         e.add_field(
-            name="Opponent:", value=f"{war.opponent.name}\n" f"{war.opponent.tag}", inline=False)
+            name="Opponent:",
+            value=f"{war.opponent.name}\n" f"{war.opponent.tag}", inline=False)
         e.add_field(name="War End Time:",
-                    value=f"{hours} hours {minutes} minutes {seconds} seconds", inline=False)
+                    value=f"{hours} hours {minutes} minutes {seconds} seconds",
+                    inline=False)
 
     await ctx.send(embed=e)
 
 
-async def run_tests_and_quit():
-    # ignore this; it is purely for the benefit of being able to run the examples as tests.
-    import sys
-
-    class Msg:
-        _state = bot._connection
+async def main():
+    logging.basicConfig(level=logging.ERROR)
 
-    await bot.wait_until_ready()
-    ctx = commands.Context(prefix=None, message=Msg, bot=bot)
+    # Instantiate the coc client used to log in
+    coc_client = coc.EventsClient(
```

Comment:
I guess something to note is that you don't have to declare this line here. We could've left coc_client = ... at the top of the file and just called .login() in this main function - that way it's referencable through the file without introducing ctx.bot.client attributes. 

Again, with all of these, just musing. Not sure if you had intentionally done this

Reply:
the style guides I have always followed is to always avoid global variables. Do you still want it as a global?